### ISSUE TITLE
Log predictions as model is training

### ISSUE THREAD

**Issue by lgvaz** [airctic/icevision#275]
## 🚀 Feature
As the model is training, it would be very helpful to visualize how the predictions are evolving.

Logging all validation set on every epoch might be too expensive, so we need a strategy for doing this.

Also, it's clear how we can log predictions use a `wandb` callback, but should we also be able to log without `wandb`?

**Comment by oke-aditya** [comment]
Logging without wandb should also be actually simple. 
1. Either we can log to tensorboard using Tesorboard callback for Lightning.
2. Simply choose 'num_samples` random samples from the dataloader, Change the model to eval mode, and get the predictions and bounding boxes. We should not be burdening the RAM by continuously storing these items into RAM. It occurs when we try to keep logging them into python. It should simply save in a folder like

`Epoch_0/Predictions/img1.jpg` -> These should have boxes drawn

Some parameters I can think of while logging.

`start_logging_after` : - Should log after `x` epochs.
`log_every` : - Every `x` number of epochs
`num_samples`: - Number of samples to log
`log_dir`: - Where to save these logging images.

Useful way will be create an API `mantisshrimp.loggers` and have these tools generic for fastai and lightning, written in pure PyTorch.

**Comment by FraPochetti** [comment]
Not relevant anymore
